### PR TITLE
Test get resource with relationships endpoint

### PR DESCRIPTION
**Pull request by yashvardhannanavati**

### CODE REVIEW

**Comment by mprahl** [comment]
@yashvardhannanavati this doesn't look like your up to date code

**Comment by yashvardhannanavati** [comment]
I really hope I haven’t accidentally lost the code😰

**Comment by yashvardhannanavati** [comment]
I’ll take a look at it first thing tomorrow morning. This isn’t my most recent code for sure

**Comment by yashvardhannanavati** [comment]
@mprahl Fixed it! :relieved:

**Comment by yashvardhannanavati** [comment]
@mprahl  Fixed the tests! Once the bug is solved, all the tests should pass :smiley:

**Comment by yashvardhannanavati** [comment]
@mprahl Based on our talk, I added the child relationship as well. Can you please review it?

File: tests/api/test_get_resource.py [hunk 1]
```diff
@@ -0,0 +1,486 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+from __future__ import unicode_literals
+import json
+from datetime import datetime
+import pytest
```
Comment (mprahl) [inline tests/api/test_get_resource.py]: Can you add a new line between `datetime` and `pytest`?

File: tests/api/test_get_resource.py [hunk 2]
```diff
@@ -0,0 +1,486 @@
...
+        'qa_contact': {
+            'email': 'jsmith@domain.local',
+            'name': None,
+            'username': 'jsmith'
+        },
+        'related_by_commits': [],
+        'reporter': {
+            'email': 'tom.brady@domain.local',
+            'name': None,
+            'username': 'tbrady'},
```
Comment (mprahl) [inline tests/api/test_get_resource.py]: Can you use the same style as you did with `qa_contact` above, where the last curly brace is on a new line? Consistency of style is desired is all.

File: tests/api/test_get_resource.py [hunk 3]
```diff
@@ -0,0 +1,486 @@
...
+                'commit_date': '2017-04-27T11:44:38+00:00',
+                'hash': '1263adb248ba633e205067e1ad6dc61931727c2d',
+                'log_message': 'Related: #12345 - fix xz'
+            },
+            {
+                'author_date': '2017-04-26T11:44:38+00:00',
+                'commit_date': '2017-04-26T11:44:38+00:00',
+                'hash': '8a63adb248ba633e200067e1ad6dc61931727bad',
+                'log_message': 'Related: #12345 - fix xyz'
+            }],
```
Comment (mprahl) [inline tests/api/test_get_resource.py]: Similar comment as on line 58.

File: tests/api/test_get_resource.py [hunk 4]
```diff
@@ -0,0 +1,486 @@
...
+        'severity': 'low',
+        'short_description': 'Some description',
+        'status': 'VERIFIED',
+        'target_milestone': 'rc',
+        'votes': 0
+    }),
+    ('distgitcommit', '8a63adb248ba633e200067e1ad6dc61931727bad', {
+        'author': {'email': 'tom.brady@domain.local',
+                   'name': None,
+                   'username': 'tbrady'},
```
Comment (mprahl) [inline tests/api/test_get_resource.py]: Similar comment as on line 58.

File: tests/api/test_get_resource.py [hunk 5]
```diff
@@ -0,0 +1,486 @@
...
+        'author_date': '2017-04-26T11:44:38+00:00',
+        'branches': [{'name': 'some_branch_name',
+                      'repo_name': 'some_repo_name',
+                      'repo_namespace': 'some_repo_namespace'}],
```
Comment (mprahl) [inline tests/api/test_get_resource.py]: Similar comment as on line 58.

File: tests/api/test_get_resource.py [hunk 6]
```diff
@@ -0,0 +1,486 @@
...
+        'children': [{'author_date': '2017-04-27T11:44:38+00:00',
+                      'commit_date': '2017-04-27T11:44:38+00:00',
+                      'hash': '5663adb248ba633e205067e1ad6dc61931727123',
+                      'log_message': 'Revert: #12345'}],
```
Comment (mprahl) [inline tests/api/test_get_resource.py]: Similar comment as on line 58.

File: tests/api/test_get_resource.py [hunk 7]
```diff
@@ -0,0 +1,486 @@
...
+        'commit_date': '2017-04-26T11:44:38+00:00',
+        'committer': {'email': 'jsmith@domain.local',
+                      'name': None,
+                      'username': 'jsmith'},
```
Comment (mprahl) [inline tests/api/test_get_resource.py]: Similar comment as on line 58.

File: tests/api/test_get_resource.py [hunk 8]
```diff
@@ -0,0 +1,486 @@
...
+        'hash': '8a63adb248ba633e200067e1ad6dc61931727bad',
+        'koji_builds': [],
+        'log_message': 'Related: #12345 - fix xyz',
+        'parent': {'author_date': '2017-04-27T11:44:38+00:00',
+                   'commit_date': '2017-04-27T11:44:38+00:00',
+                   'hash': '1263adb248ba633e205067e1ad6dc61931727c2d',
+                   'log_message': 'Related: #12345 - fix xz'},
```
Comment (mprahl) [inline tests/api/test_get_resource.py]: Similar comment as on line 58.

File: tests/api/test_get_resource.py [hunk 9]
```diff
@@ -0,0 +1,486 @@
...
+        'pushes': [{'id': '56789',
+                    'push_date': '2017-08-01T15:43:51+00:00',
+                    'push_ip': '28.03.19.94'}],
```
Comment (mprahl) [inline tests/api/test_get_resource.py]: Similar comment as on line 58.

File: tests/api/test_get_resource.py [hunk 10]
```diff
@@ -0,0 +1,486 @@
...
+        'related_bugs': [{'classification': 'Red Hat',
```
Comment (mprahl) [inline tests/api/test_get_resource.py]: Can you have this in the same style that you did on line 60?

File: tests/api/test_get_resource.py [hunk 11]
```diff
@@ -0,0 +1,486 @@
...
+                          'product_name': 'Red Hat Enterprise Linux',
+                          'product_version': '7.5',
+                          'resolution': '',
+                          'severity': 'low',
+                          'short_description': 'Some description',
+                          'status': 'VERIFIED',
+                          'target_milestone': 'rc',
+                          'votes': 0}],
+        'repos': [{'name': 'some_repo', 'namespace': 'some_namespace'}],
+        'resolved_bugs': [{'classification': 'Red Hat',
```
Comment (mprahl) [inline tests/api/test_get_resource.py]: Can you have this in the same style that you did on line 60?

File: tests/api/test_get_resource.py [hunk 12]
```diff
@@ -0,0 +1,486 @@
...
+        commit_two.resolved_bugs.connect(bug)
+        commit_three.reverted_bugs.connect(bug)
+        advisory.attached_bugs.connect(bug)
+
+    if resource == 'distgitcommit':
+        commit.author.connect(tbrady)
+        commit.committer.connect(jsmith)
+        commit.parent.connect(commit_two)
+        commit_three.parent.connect(commit)
+        commit.related_bugs.connect(bug)
```
Comment (mprahl) [inline tests/api/test_get_resource.py]: The relationships are all with the same bug. Can you create other ones to test that the relationships are correct?

File: tests/api/test_get_resource.py [hunk 13]
```diff
@@ -0,0 +1,486 @@
...
+                           'priority': 'high',
+                           'product_name': 'Red Hat Enterprise Linux',
+                           'product_version': '7.5',
+                           'resolution': '',
+                           'severity': 'low',
+                           'short_description': 'Some description',
+                           'status': 'VERIFIED',
+                           'target_milestone': 'rc',
+                           'votes': 0}],
+        'reverted_bugs': [{'classification': 'Red Hat',
```
Comment (mprahl) [inline tests/api/test_get_resource.py]: Can you have this in the same style that you did on line 60?

File: tests/api/test_get_resource.py [hunk 14]
```diff
@@ -0,0 +1,486 @@
...
+                           'resolution': '',
+                           'severity': 'low',
+                           'short_description': 'Some description',
+                           'status': 'VERIFIED',
+                           'target_milestone': 'rc',
+                           'votes': 0}]
+    }),
+    ('kojibuild', '2345', {
+        'advisories': [],
+        'commit': {'author_date': '2017-04-27T11:44:38+00:00',
```
Comment (mprahl) [inline tests/api/test_get_resource.py]: Similar comment as on line 58.

File: tests/api/test_get_resource.py [hunk 15]
```diff
@@ -0,0 +1,486 @@
...
+                   'log_message': 'Related: #12345 - fix xz'},
+        'completion_time': '2017-04-02T19:39:06+00:00',
+        'creation_time': '2017-04-02T19:39:06+00:00',
+        'epoch': '0',
+        'extra': None,
+        'id': '2345',
+        'name': 'slf4j',
+        'owner': {'email': 'matt.prahl@domain.local',
+                  'name': None,
+                  'username': 'mprahl'},
```
Comment (mprahl) [inline tests/api/test_get_resource.py]: Similar comment as on line 58.

File: tests/api/test_get_resource.py [hunk 16]
```diff
@@ -0,0 +1,486 @@
...
+        'release': '4.el7_4',
+        'start_time': '2017-04-02T19:39:06+00:00',
+        'state': 1,
+        'tags': [{'id': '2702', 'name': 'some_active_tag'}],
+        'tasks': [{'arch': 'no_arch',
```
Comment (mprahl) [inline tests/api/test_get_resource.py]: Can you have this in the same style that you did on line 60?

File: tests/api/test_get_resource.py [hunk 17]
```diff
@@ -0,0 +1,486 @@
...
+                   'weight': 123.45}],
+        'version': '1.7.4'
+    }),
+    ('advisory', '27825', {
+        'actual_ship_date': '2017-08-01T15:43:51+00:00',
+        'advisory_name': 'RHBA-2017:2251-02',
+        'assigned_to': {'email': 'matt.prahl@domain.local',
+                        'name': None,
+                        'username': 'mprahl'},
+        'attached_bugs': [{'classification': 'Red Hat',
```
Comment (mprahl) [inline tests/api/test_get_resource.py]: Can you have this in the same style that you did on line 60?

File: tests/api/test_get_resource.py [hunk 18]
```diff
@@ -0,0 +1,486 @@
...
+                           'priority': 'high',
+                           'product_name': 'Red Hat Enterprise Linux',
+                           'product_version': '7.5',
+                           'resolution': '',
+                           'severity': 'low',
+                           'short_description': 'Some description',
+                           'status': 'VERIFIED',
+                           'target_milestone': 'rc',
+                           'votes': 0}],
+        'attached_builds': [{'completion_time': '2017-04-02T19:39:06+00:00',
```
Comment (mprahl) [inline tests/api/test_get_resource.py]: Can you have this in the same style that you did on line 60?

File: tests/api/test_get_resource.py [hunk 19]
```diff
@@ -0,0 +1,486 @@
...
+                                  'release_date': None,
+                                  'security_impact': 'None',
+                                  'security_sla': None,
+                                  'state': 'SHIPPED_LIVE',
+                                  'status_time': '2017-08-01T15:43:51+00:00',
+                                  'synopsis': 'cifs-utils bug fix update',
+                                  'type': 'RHBA',
+                                  'update_date': '2017-08-01T07:16:00+00:00',
+                                  'updated_at': '2017-08-01T15:43:51+00:00'},
+        'triggered_container_builds': [{'build_id': 15639047,
```
Comment (mprahl) [inline tests/api/test_get_resource.py]: Can you have this in the same style that you did on line 60?

File: tests/api/test_get_resource.py [hunk 20]
```diff
@@ -0,0 +1,486 @@
...
+                             'start_time': '2017-04-02T19:39:06+00:00',
+                             'state': 1,
+                             'version': '1.7.4'}],
+        'content_types': ['docker'],
+        'created_at': '2017-04-03T14:47:23+00:00',
+        'id': '27825',
+        'issue_date': '2017-08-01T05:59:34+00:00',
+        'package_owner': {'email': 'tom.brady@domain.local',
+                          'name': None,
+                          'username': 'tbrady'},
```
Comment (mprahl) [inline tests/api/test_get_resource.py]: Similar comment as on line 58.

File: tests/api/test_get_resource.py [hunk 21]
```diff
@@ -0,0 +1,486 @@
...
+        'product_name': 'Red Hat Enterprise Linux',
+        'product_short_name': 'RHEL',
+        'release_date': None,
+        'reporter': {'email': 'jsmith@domain.local',
```
Comment (mprahl) [inline tests/api/test_get_resource.py]: Similar comment as on line 58.

File: tests/api/test_get_resource.py [hunk 22]
```diff
@@ -0,0 +1,486 @@
...
+                     'name': None,
+                     'username': 'jsmith'},
+        'security_impact': 'None',
+        'security_sla': None,
+        'state': 'SHIPPED_LIVE',
+        'states': [{'created_at': '2017-04-02T19:45:06+00:00',
```
Comment (mprahl) [inline tests/api/test_get_resource.py]: Can you have this in the same style that you did on line 60?

File: tests/api/test_get_resource.py [hunk 23]
```diff
@@ -0,0 +1,486 @@
...
+        'id': '397',
+        'name': 'jboss-eap-7-eap70-openshift-docker',
+        'original_nvr': 'jboss-eap-7-eap70-openshift-test',
+        'rebuilt_nvr': 'jboss-eap-7-eap70-openshift-test',
+        'state': 1,
+        'state_name': 'DONE',
+        'state_reason': 'Built successfully.',
+        'time_completed': '2017-04-02T19:39:06+00:00',
+        'time_submitted': '2017-04-02T19:39:06+00:00',
+        'triggered_by_freshmaker_event': {'event_type_id': 8,
```
Comment (mprahl) [inline tests/api/test_get_resource.py]: Similar comment as on line 58.

File: tests/api/test_get_resource.py [hunk 24]
```diff
@@ -0,0 +1,486 @@
...
+        'updated_at': datetime(2017, 8, 1, 15, 43, 51)
+    })[0]
+    adv_state = AdvisoryState.get_or_create({
+        'id_': '7890',
+        'created_at': datetime(2017, 4, 2, 19, 45, 6),
+        'updated_at': datetime(2017, 4, 2, 19, 45, 6),
+        'name': 'some_name'
+    })[0]
+    fm_event = FreshmakerEvent.get_or_create({
+        "event_type_id": 8,
```
Comment (mprahl) [inline tests/api/test_get_resource.py]: Please use single quotes for consistency across the code-base.

File: tests/api/test_get_resource.py [hunk 25]
```diff
@@ -0,0 +1,486 @@
...
+    if resource == 'freshmakerevent':
+        fm_event.conditional_connect(fm_event.triggered_by_advisory, advisory)
+        fm_event.triggered_container_builds.connect(cb)
+
+    if resource == 'containerbuilds':
+        fm_event.triggered_container_builds.connect(cb)
+
+    rv = client.get('/api/v1/{0}/{1}'.format(resource, uid))
+    assert rv.status_code == 200
+    import pprint
```
Comment (mprahl) [inline tests/api/test_get_resource.py]: Can you delete these `pprint` lines?

File: tests/api/test_get_resource.py [hunk 26]
```diff
@@ -0,0 +1,641 @@
...
+        'modified_time':'2018-02-07T19:30:47+00:00',
+        'priority':'high',
+        'product_name':'Red Hat Enterprise Linux',
+        'product_version':'7.5',
+        'qa_contact':{
+            'email': 'jsmith@domain.local',
+            'name': None,
+            'username': 'jsmith'
+        },
+        'related_by_commits': [
```
Comment (mprahl) [inline tests/api/test_get_resource.py]: Bike-shedding: Empty list can be in one line.

File: tests/api/test_get_resource.py [hunk 27]
```diff
@@ -0,0 +1,641 @@
...
+        'assignee': {
+            'email': 'matt.prahl@domain.local',
+            'name': None,
+            'username': 'mprahl'
+        },
+        'attached_advisories': [
+            {
+                'actual_ship_date': '2017-08-01T15:43:51+00:00',
+                'advisory_name': 'RHBA-2017:2251-02',
+                'content_types': [
```
Comment (mprahl) [inline tests/api/test_get_resource.py]: Bike-shedding: Single elements can be on one line

File: tests/api/test_get_resource.py [hunk 28]
```diff
@@ -0,0 +1,641 @@
...
+                'product_name': 'Satellite',
+                'product_version': '3',
+                'resolution': '',
+                'severity': 'medium',
+                'short_description': 'Some description',
+                'status': 'VERIFIED',
+                'target_milestone': 'rc',
+                'votes': 0
+            },
+            {
```
Comment (mprahl) [inline tests/api/test_get_resource.py]: This bug is repeated in resolved bugs. This can be removed  and disconnected in the code.

Comment (mprahl) [reply tests/api/test_get_resource.py]: NVM, this is fine for test data.

**Comment by mprahl** [comment]
:+1

**Pull request merged**